### ISSUE TITLE
Odd warnings and errors in Pilot log when Gateway deleted

### ISSUE THREAD

**Issue by esnible** [istio/istio#12495]
Verified on 1.1 RC 4 and 5; I have not tried with other versions.

To reproduce, create a Gateway and start `kubectl logs -n istio-system <pilot-pod> -c discovery -f`.  The Gateway I used was

```
cat <<EOF | kubectl create -f -
apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  name: sm-test
  namespace: default
spec:
  selector:
    istio: ingressgateway # use istio default controller
  servers:
  - port:
      number: 20433
      name: http
      protocol: HTTP
    hosts:
    - "*"
EOF
```

Then delete the gateway: `kubectl delete gateway sm-test`

The Pilot discovery logs fill with strange errors and warnings of this format:
```
2019-03-14T15:52:32.410617Z	info	ads	LDS: PUSH for node:istio-ingressgateway-7df56686c4-68s8c.istio-system addr:"172.30.150.248:33408" listeners:1 1063
2019-03-14T15:52:32.412426Z	error	buildGatewayRoutes: could not find server for routeName http.20433, have map[http.80:[port:<number:80 protocol:"HTTP" name:"http" > hosts:"*" ]]
2019-03-14T15:52:32.412483Z	warn	ads	RDS: Failed to generate routes for route http.20433 for node &{Kubernetes router [172.30.150.248] istio-ingressgateway-7df56686c4-68s8c.istio-system region:"us-south" zone:"dal13"  istio-system.svc.cluster.local istio-system  map[ROUTER_MODE:sni-dnat ISTIO_META_INSTANCE_IPS:172.30.150.248,172.30.150.248,fe80::e835:66ff:fe21:8be3 POD_NAME:istio-ingressgateway-7df56686c4-68s8c istio:sidecar ISTIO_PROXY_VERSION:1.1.0 CONFIG_NAMESPACE:istio-system ISTIO_PROXY_SHA:istio-proxy:8ddb2aab27a5d311adf81995f998df62e4c0e885 ISTIO_VERSION:1.1.0-rc.4] <nil> [0xc4209cd100 0xc4209cd200 0xc4209cd300 0xc4209cd480 0xc4209cd580 0xc4209cd680 0xc4209cd780 0xc4209cd880 0xc4209cd980]}: buildGatewayRoutes: could not find server for routeName http.20433, have map[http.80:[port:<number:80 protocol:"HTTP" name:"http" > hosts:"*" ]]
2019-03-14T15:52:32.412859Z	info	ads	ADS: "172.30.150.248:33408" router~172.30.150.248~istio-ingressgateway-7df56686c4-68s8c.istio-system~istio-system.svc.cluster.local-181 terminated rpc error: code = Canceled desc = context canceled
```

I saw 352 of these buildGatewayRoutes errors on a cluster that just had a small Istio control plane and a bookinfo.  I suspect there would have been more errors on a heavily loaded cluster.

**Comment by esnible** [comment]
Still a problem under 1.1.1.

**Comment by howardjohn** [comment]
Fixed in 1.1.6 https://github.com/istio/istio/pull/13924
